src/core/fem/src/nurbs_discretization/4C_fem_nurbs_discretization_knotvector.cpp

```
@@ -1038,4 +1029,62 @@ void Core::FE::Nurbs::Knotvector::print(std::ostream& os) const
   }
 }
 
+Core::IO::InputSpec Core::FE::Nurbs::Knotvector::spec()
+{
+  using namespace Core::IO::InputSpecBuilders;
+  using namespace Core::IO::InputSpecBuilders::Validators;
+
+  return all_of({
+      parameter<int>("NURBS_DIMENSION",
+          {.description = "Spatial dimension of the NURBS patch", .validator = positive<int>()}),
+      list("PATCHES",
+          all_of({
+              parameter<int>(
+                  "ID", {.description = "ID of the NURBS patch", .validator = positive<int>()}),
+              list("KNOT_VECTORS",
+                  all_of({
+                      parameter<int>("DEGREE", {.description = "Degree of the NURBS interpolation",
+                                                   .validator = positive<int>()}),
+                      parameter<KnotvectorType>(
+                          "TYPE", {.description = "Type of the knot vector",
+                                      .default_value = KnotvectorType::Interpolated}),
+                      parameter<std::vector<double>>(
+                          "KNOTS", {.description = "Support points of the NURSB interpolation"}),
+                  })),
+          })),
+  });
+}
+
+
+Core::FE::Nurbs::Knotvector Core::FE::Nurbs::Knotvector::from_input(
+    const Core::IO::InputParameterContainer& data)
+{
+  unsigned dim = data.get<int>("NURBS_DIMENSION");
```

I think we should not specify a fixed NURBS dimension in the input file, but instead allow patches to have different dimensions.

The dimension of each NURBS patch should simply be determined by the number of knot vectors it contains.

It looks like we currently have some issues when reading patches of different sizes, but in principle the code itself should handle this. For now we should simply throw an error when inconsistent patch dimensions are found (i.e., not all have the same dimension) - but avoid having a global NURBS dimension.

@eulovi and myself can have a look on how to then get that working in the code in the future (should be a simple fix).